### ISSUE TITLE
Use `bin/rails` rather than `rake` for deployment tasks

### ISSUE THREAD

**Issue by jcoyne** [capistrano/rails#253]
Because this is the documented way of running migrations https://guides.rubyonrails.org/active_record_migrations.html#running-migrations

and this runs the migrations in the current bundle context if they are using bundler, without any extra capistrano gems required.

**Comment by mathagician** [comment]
@jcoyne  Is it a suggestion or any code changes is required? can u guide ?

**Comment by jcoyne** [comment]
Yes, this was a feature request.  I'm pretty sure the existing command is here:
https://github.com/capistrano/rails/blob/6c8603f69267eb173eb8f2edd0a1329f1c50fef5/lib/capistrano/tasks/migrations.rake#L25

but I'm not sure how to replace `:rake` with `bin/rails` as I'm not sure that method accepts non-Symbol arguments.

**Comment by matt17r** [comment]
I stumbled across this issue after executing a remote rake task with Capistrano and being warned that I'll need to execute it with `bin/rails` from Rails 8.1 onwards.

My current Capistrano task is:
```
desc "Invoke arbitrary rake tasks on the server"
task :invoke, [:command] => "deploy:set_rails_env" do |task, args|
  on primary(:app) do
    within current_path do
      with :rails_env => fetch(:rails_env) do
        rake args[:command]
      end
    end
  end
end
```

I naively thought it might be as simple as switching to `execute` and calling _bin/rails_ directly with `execute "bin/rails #{args[:command]}"`

...but that fails because it can't find _bin/rails_. I've since learned from the [SSHKit Readme](https://github.com/capistrano/sshkit?tab=readme-ov-file#running-commands) that this is expected behaviour (`within` and `with` don't work when the `execute` command is a string instead of a symbol).

This doesn't address the OP's question/suggestion of switching to `bin/rails` for everything but I am now able to run a one off command using rake **or** bin/rails (via bundle exec*) using:
```
namespace :invoke do
  desc "Invoke arbitrary rake task(s) on the server"
  task :rake, [:command] => "deploy:set_rails_env" do |task, args|
    on primary(:app) do
      within current_path do
        with :rails_env => fetch(:rails_env) do
          rake args[:command]
        end
      end
    end
  end

  desc "Invoke an arbitrary bin/rails command on the server"
  task :rails, [:command] => "deploy:set_rails_env" do |task, args|
    on primary(:app) do
      within current_path do
        with :rails_env => fetch(:rails_env) do
          execute :bundle, :exec, :rails, args[:command]
        end
      end
    end
  end
end
```

*According to [this (fairly old) StackOverflow answer](https://stackoverflow.com/a/52526163), `bundle exec` eventually calls the rails binstub. The last path in that chain hasn't changed so I'm assuming this answer is still valid for Rails 8
